### ISSUE TITLE
There is an extremely serious problem with the replacement function!!!!!!

### ISSUE THREAD

**Issue by FakerDomain** [godotengine/godot#91406]
### Tested versions

4.2.2 stable

### System information

Windows10

### Issue description

When I use the ctrl+alt+F to replace a variable's name, some of the code I wrote before using the replace function in the currently open script disappears directly

### Steps to reproduce

1. create a script, and add some code like below:
`extends Node

var g_test: int
var b_test: int

func start() -> void:
      if g_test:
      print(g_test)`

2. write some additional code below the 'star()' function, any code (or comments), it doesn't matter, 
3. press ctrl+alt+F to replace for example, 'g_test' with 'a_test' and chose to replace them all.
4. then, what you just write in the 2th step will disappear~

### Minimal reproduction project (MRP)

N/A

**Comment by FakerDomain** [comment]
> When you say CTRL+ALT+F, do you mean "Replace in Files"? Because that is bound to CTRL+SHIFT+R by default, with CTRL+ALT+F bringing up the "Jump to Function" modal, which doesn't seem to allow for any replacements.
> 
> Regardless, I was able to reproduce the issue with "Replace in Files". As the name implies, it looks for the provided text in the actual source files, while "Replace" looks at the contents of the text editor. This means that any replacement done with "Replace in Files" is going to result in a reload request from editor as the file is newer on disk. If you have any unsaved changes in the editor, they will be lost in the reload.
> 
> Can you confirm that the issue doesn't present itself if you save all changes before executing the replacement routine?

Yeah, you are right, I mean Ctrl+Shift+F. And I tried what you mentioned, my code won't disappear anymore if I saved the script before I replace something. However, I still wish my script can be saved automatically when I use the replacement function, since last time my code (a lot of code) disappeared, which made me very frustrated!

**Comment by ArchieVillain** [comment]
You may want to throw your support behind this proposal: godotengine/godot-proposals#9654.
